controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1574,7 +1574,7 @@ int bml_internal::get_dcs_scan_results(const sMacAddr &mac, BML_NEIGHBOR_AP *res
                                        const unsigned int max_results_size, uint8_t &result_status,
                                        bool is_single_scan)
 {
-    if (!results || !results_size || !result_status) {
+    if (!results || max_results_size == 0) {
```

Comment:
I understand and agree with your point that `!results` is an  input parameter validity check
But I think that `max_results_size == 0` could also be classified as a parameter validity check since it check the validity of the parameter `max_results_size`

Since the parameter `max_results_size` is defined as a `const unsigned int` is cannot have a signed value, but can still be set as 0, which is invalid for the "get results" flow since it would mean that the client has requested 0 results
therefore the error `"Function is called, but no data is being requested!"`

Reply:
OK fair enough.